wdl-lint/src/rules/nonmatching_output.rs

```diff
@@ -75,30 +59,20 @@ fn extra_output_in_meta(span: Span, name: &str, context: &TaskOrWorkflow) -> Dia
 }
 
 /// Creates a diagnostic for out-of-order entries.
-fn out_of_order(span: Span, output_span: Span, context: &TaskOrWorkflow) -> Diagnostic {
-    let (ty, item_name) = match context {
-        TaskOrWorkflow::Task(t) => ("task", t.name().as_str().to_string()),
-        TaskOrWorkflow::Workflow(w) => ("workflow", w.name().as_str().to_string()),
-    };
-
+fn out_of_order(span: Span, output_span: Span, item_name: String, ty: &str) -> Diagnostic {
     Diagnostic::warning(format!(
         "`outputs` section of `meta` for the {ty} `{item_name}` is out of order"
     ))
     .with_rule(ID)
     .with_highlight(span)
     .with_highlight(output_span)
     .with_fix(
-        "ensure the keys within `meta.outputs` have the same order as they appear in `outputs`",
+        "ensure the keys within `meta.outputs` have the same order as they appear in `output`",
     )
 }
 
 /// Creates a diagnostic for non-object `meta.outputs` entries.
-fn non_object_meta_outputs(span: Span, context: &TaskOrWorkflow) -> Diagnostic {
-    let (ty, item_name) = match context {
-        TaskOrWorkflow::Task(t) => ("task", t.name().as_str().to_string()),
-        TaskOrWorkflow::Workflow(w) => ("workflow", w.name().as_str().to_string()),
-    };
-
+fn non_object_meta_outputs(span: Span, item_name: String, ty: &str) -> Diagnostic {
     Diagnostic::warning(format!(
         "`outputs` key in `meta` section for the {ty} `{item_name}` is not an object"
```

Comment:
This strikes me as potentially confusing for a user. The most likely scenario for this diagnostic (in my estimation) is that the user is using the `outputs` key with some different convention than what we expect. They look at this message here and it wouldn't make sense to them. I'd suggest changing to something along the lines of:
```suggestion
        "`outputs` key in `meta` section  is reserved for an object with keys corresponding to declared `output` values"
```
And as an aside, isn't `item_name` always going to be `outputs`?

Reply:
`item_name` has either the `task` or `workflow` name.

Reply:
@a-frantz I merged the new and old language. It's a bit wordy in the lint messages, but I think it's clear now.